src/DeviceInterfaceValues.tsx

```diff
@@ -114,18 +114,50 @@ interface ObjectDatastreamTableProps {
 const ObjectDatastreamTable = ({
   dataTreeNode,
 }: ObjectDatastreamTableProps): React.ReactElement => {
+  const [fromTime, setFromTime] = useState<string>('');
+  const [toTime, setToTime] = useState<string>('');
   const treeData = dataTreeNode.toData();
   if (treeData.length === 0) {
     return <p>No data sent by the device.</p>;
   }
 
-  const objectProperties = Object.keys(treeData[0].value);
-  const orderedData = _.orderBy(treeData, 'timestamp', 'desc');
+  const handleFromTimeChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setFromTime(e.target.value);
+  };
+
+  const handleToTimeChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setToTime(e.target.value);
+  };
+
+  const filteredData = treeData.filter((data) => {
```

Comment:
The filter has to be sent to Astarte, not filtered locally.
This needs to allow users to peek into older data.
Also, this only makes sense for datastream interfaces, as properties have no concept of older values